packages/aws-cdk-lib/aws-ec2/lib/vpc.ts

```diff
@@ -1450,14 +1450,15 @@ export class Vpc extends VpcBase {
     Tags.of(this).add(NAME_TAG, props.vpcName || this.node.path);
 
     if (props.availabilityZones) {
-      // If given AZs and stack AZs are both resolved, then validate their compatibility.
-      const resolvedStackAzs = stack.availabilityZones.filter(az => !Token.isUnresolved(az));
```

Comment:
The call to `stack.availabilityZones` is what ensures the CLI will perform the lookup, if you remove this, what guarantees the context will be created?

Reply:
Thanks for pointing this out. I didn't catch this in the code. With that being said, this should obviously be re-included in the code to ensure the context will be created.